wordpress-mobile/WordPress-Android
# Login rework: Username/Password screen for selfhosted

This PR introduces the username+password screen for the selfhosted flow. It supports both selfhosted and WPCOM sites.

The final visual touches, the messages and final texts will be done in a future PR.

To test:
* With the app signed out or its data cleaned, launch the app
* Tap on the "Login" button to go to the email screen
* Tap on the "Log in to your site by entering your site address instead." link on the lower left to go to the site address input screen
* Enter the URL of your selfhosted or wpcom site and tap on "NEXT"
* Enter your username and password in the respective input fields and tap on "NEXT"
* Notice the app do some work (progress popup) and jumping to the epilogue screen on success
